### PR TITLE
Add `Clear` helper trait, `Map`-like wrapper for tick -> state

### PR DESCRIPTION
**Pull request by MingweiSamuel**

### CODE REVIEW

**Comment by justinj** [comment]
I like this, my take on the persistence stuff for now is [here](https://github.com/hydro-project/hydroflow/issues/103#issuecomment-1090837674) and that we should go ahead with this approach.

**Comment by MingweiSamuel** [comment]
Maybe use `scratch` naming

**Comment by jhellerstein** [comment]
Just don't make it be tied to a type called "tick" or any other assumption
that there's an agreed-upon temporal model. Let that be external, and here
we just have nonce->state mapping.

J

**Comment by MingweiSamuel** [comment]
Starting with a very simple wrapper and API for now, will probably change as we write use cases

File: hydroflow/src/lang/monotonic_map.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+impl<K, V> MonotonicMap<K, V>
+where
+    K: PartialOrd,
+    V: Clear,
+{
+    pub fn new_init(val: V) -> Self {
+        Self { key: None, val }
+    }
+
+    /// Returns the value for the monotonicly increasing key, or `None` if the
```
Comment (justinj) [inline hydroflow/src/lang/monotonic_map.rs:35]: nit: `monotonically` i think?

File: hydroflow/src/lang/monotonic_map.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
+use super::clear::Clear;
+
```
Comment (justinj) [inline hydroflow/src/lang/monotonic_map.rs:2]: nit: doesn't have to be in this PR but a comment or example of where this thing is useful would be good

**Review by justinj — APPROVED** [review]

File: hydroflow/src/lang/monotonic_map.rs (hunk 1 above)
Comment (MingweiSamuel) [reply hydroflow/src/lang/monotonic_map.rs:35]: 👌

**Pull request merged**